extensions/azdata/src/extension.ts

```diff
@@ -5,19 +5,44 @@
 
 import * as azdataExt from 'azdata-ext';
 import * as vscode from 'vscode';
-import { checkAndUpgradeAzdata, findAzdata, IAzdataTool } from './azdata';
+import { checkAndInstallAzdata, checkAndUpgradeAzdata, findAzdata, IAzdataTool } from './azdata';
+import Logger from './common/logger';
 import * as loc from './localizedConstants';
 
 let localAzdata: IAzdataTool | undefined = undefined;
 
 export async function activate(): Promise<azdataExt.IExtension> {
-	localAzdata = await checkForAzdata();
+	await vscode.commands.executeCommand('setContext', 'config.deployment.azdataFound', false);
+
+	vscode.commands.registerCommand('azdata.install', async () => {
+		await checkAndInstallAzdata(true);
+	});
+
+	vscode.commands.registerCommand('azdata.upgrade', async () => {
+		const currentAzdata = await findAzdata();
+		if (currentAzdata !== undefined) {
+			await checkAndUpgradeAzdata(currentAzdata, true);
```

Comment:
You should be updating localAzdata after this.

And really you should either be using localAzdata or updating it even after running the findAzdata here.

(this isn't the greatest design so it's a bit of a pain right now - I plan on revisiting this at some point to make keeping track of the known version of azdata easier. But for now let's just work with this)

Reply:
done.